src/http/one/Tokenizer.h

```diff
@@ -9,68 +9,46 @@
 #ifndef SQUID_SRC_HTTP_ONE_TOKENIZER_H
 #define SQUID_SRC_HTTP_ONE_TOKENIZER_H
 
-#include "parser/Tokenizer.h"
+class SBuf;
+
+namespace Parser {
+class Tokenizer;
+}
 
 namespace Http {
 namespace One {
 
 /**
- * Lexical processor extended to tokenize HTTP/1.x syntax.
+ * Attempt to parse a (token / quoted-string ) lexical construct.
```

Comment:
I think we should simplify and detail this, especially the important aspect of the token prefix being indistinguishable from the complete token (at this function scope). For example:
```suggestion
 * Extracts either an HTTP/1 token (prefix) or a complete HTTP/1 quoted-string.
 * Unescapes escaped characters in HTTP/1.1 quoted strings.
 * The caller can (and must) distinguish token prefixes from complete tokens. This
 * is usually done by checking for the trailing garbage and/or expression terminators.
```

Reply:
I am resolving this conversation. The problem with the new documentation and implementation is discussed in another (unresolved) https://github.com/squid-cache/squid/pull/310#discussion_r228577286